RESTserver/src/main/java/com/softserve/if072/restservice/controller/CategoryController.java

```diff
@@ -1,58 +1,85 @@
 package com.softserve.if072.restservice.controller;
 
 import com.softserve.if072.common.model.Category;
+import com.softserve.if072.restservice.exception.DataNotFoundException;
 import com.softserve.if072.restservice.service.CategoryService;
+import org.apache.logging.log4j.LogManager;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.*;
+import org.apache.logging.log4j.Logger;
 
+import javax.servlet.http.HttpServletResponse;
 import java.util.List;
 
 @Controller
 @RequestMapping(value = "/category")
+@PropertySource("classpath:message.properties")
 public class CategoryController {
 
-    @Autowired
+    private static final Logger LOGGER = LogManager.getLogger(CategoryController.class);
     private CategoryService categoryService;
 
-    @RequestMapping(value = "/all", method = RequestMethod.GET)
-    @ResponseBody
-    @ResponseStatus(value = HttpStatus.OK)
-    public List<Category> getAll() {
-        return categoryService.getAll();
+    @Autowired
+    public CategoryController(CategoryService categoryService) {
+        this.categoryService = categoryService;
     }
 
-    @RequestMapping(value = "/get/{id}", method = RequestMethod.GET)
+    @Value("${category.notFound}")
+    private String categoryNotFound;
+
+    @GetMapping(value = "/user/{userID}")
     @ResponseBody
     @ResponseStatus(value = HttpStatus.OK)
-    public Category getById(@PathVariable("id") int id) {
-        return categoryService.getById(id);
+    public List<Category> getAllCategoriesByUserID(@PathVariable("userID") int userID, HttpServletResponse response) {
+        try {
+            List<Category> categories = categoryService.getByUserID(userID);
+            LOGGER.info("All categories were found");
+            return categories;
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error("Categories were not found", e);
```

Comment:
No need to create this message here. you can get error message from the exception, by `e.getMessage()`